zerver/tests/test_realm.py

```diff
@@ -898,6 +904,39 @@ def test_update_realm_allow_message_editing(self) -> None:
         self.assertEqual(realm.message_content_edit_limit_seconds, 200)
         self.assertEqual(realm.allow_community_topic_editing, False)
 
+    def test_prevent_realm_public_listing_if_plan_type_is_not_standard_free(self) -> None:
+        realm = get_realm("zulip")
+        realm.oss_showcase_policy = 1
+        realm.save(update_fields=["oss_showcase_policy"])
+        self.assertEqual(realm.get_plan_type, Realm.SELF_HOSTED)
+        result = self.client_patch("/json/realm", {"oss_showcase_policy": "2"})
+        self.assertEqual(result.status_code, 400)
+        substring = "Your organization does not qualify to enable this feature."
+        self.assert_in_response(substring, result)
+
+    def test_realm_ready_for_public_listing_when_conditions_are_satisfied(self) -> None:
+        """
+        When the realm's plan type is STANDARD_FREE and OSS_SHOWCASE is set
+        as True in settings. We should be able to update the oss_showcase_policy.
+        """
+        realm = get_realm("zulip")
+        realm.plan_type = Realm.STANDARD_FREE
+        realm.save(update_fields=["plan_type"])
```

Comment:
Please use `do_change_plan_type` for making these data setup changes, since it has side effects and we'd like to test realistic situations.